global-cli/index.js

```diff
@@ -167,6 +170,29 @@ function checkNodeVersion() {
   }
 }
 
+function checkAppName(appName) {
+  // TODO: there should be a single place that holds the dependencies
+  var dependencies = ['react', 'react-dom'];
+  var devDependencies = ['react-scripts'];
+  var allDependencies = dependencies.concat(devDependencies).sort();
+
+  if (allDependencies.indexOf(appName) >= 0) {
+    console.error(
+      chalk.red(
+        `Can't use "${appName}" as the app name because a dependency with the same name exists.\n\n` +
```

Comment:
@mareksuscak The requirement for Node >= 4 is correct, however we want this entry point file to still run in older versions, so that instead of just crashing, we can show a warning, if someone tries to use an unsupported version.

This is what it looks like:
<img width="742" alt="screen shot 2016-09-18 at 15 16 00" src="https://cloud.githubusercontent.com/assets/497214/18615591/dcd893d0-7db2-11e6-9794-f00a9fab693d.png">